### DISCUSSION TITLE
Management of multilingual translations

### DISCUSSION THREAD

**Discussion by valentincalomme** [fastapi/fastapi#10872] [Questions]
I really admire how many languages support the FastAPI documentation. I was curious about the overall process and philosophy behind it.

I checked out the documentation, and there are already a lot of automation and ways to manage this process, but I have some more questions/suggestions.

- Assuming that the english docs are leading, and they are changed. How do other languages get updated?
- Are there dedicated language maintainers? For instance to offload the review process.
- Are pull requests automatically created when a change is introduced? For instance, `README.md` is changed and PRs are created for all supported languages for that file?
- Are there sanity checks run in the build? For instance, language detection to avoid silly suggestions, or simply length and content checks (i.e. are all links and images included, is the content roughly the same length)
- Is English always leading? Let's say I find something unclear in the French translation and I make a PR for it. How does it work with the English version?
- Can a language support some but not all of the content? It might be helpful to have some pages be translated (i.e. `README.md`) and some remain untranslated.
- I'd like to contribute to the Dutch translations, I've seen a few PRs but it's not yet officially supported on `main`, how to proceed?

All in all, I am a big fan of this project!

**Comment by hasansezertasan** [comment]
I was thinking of an automation script. A script that tracks the changes made in the `/docs/en/docs` folder and sends messages to all `Language Translation` discussions to let translation contributors know which translation file is outdated.

Example Scenario: When `docs/en/docs/index.md` has a merged commit, a message can be sent to #9193 (and other Language Discussions).

Example message:

```markdown
The `docs/en/docs/index.md` is updated and `docs/tr/docs/index.md` is outdated now.
```

Or maybe include the pull request number like...

```markdown
The `docs/en/docs/index.md` is updated (#10442) and `docs/tr/docs/index.md` (#10442) is outdated now.
```

> The PRs are the same but still...

With this script, we can keep track of outdated files.

**Reply by valentincalomme** [reply]
That is along the lines of what I was thinking indeed. Pick a leading source and notify other languages of what needs to be updated.

In the grand scheme of things, such a "diff" capability could also show the progress or the level of completeness of the docs. And perhaps warn users that a translation might be outdated?

Follow-up question. With almost 20 languages, would it make sense to split the documentation, at least the translations, to a different repo? To keep the releases and their content easier to manage? I might be trying to fix something that isn't broken, but thought I'd bring it up

**Reply by hasansezertasan** [reply]
I believe we can use the `diff` capability but there is not a "user" for any language, at least that's what I know. So warning a user wouldn't work but sending a message to a related discussion could be good enough.

Also, I don't think calculating "level of completeness" with `diff` also wouldn't work. Even do phrases change, the statement might not. There are PRs for typos and some other things that are not related to translation.

I don't have anything to say about your follow-up question. It's up to @tiangolo.

**Comment by hasansezertasan** [comment]
I have referenced this discussion today at https://github.com/reflex-dev/reflex/issues/1968#issuecomment-1924767246 I didn't want to copy and paste the comment but the people at reflex will try to come up with a solution for that, excited to hear more about it.

**Comment by YuriiMotov** [comment]
Current state is described here: https://fastapi.tiangolo.com/contributing/#translations
